packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,242 @@
+        break;
+      case Cardano.CertificateType.PoolRegistration:
+        if (rewardAccounts.some((acct) => cert.poolParameters.owners.includes(acct)))
+          deposit += depositParams.poolDeposit;
+        break;
+      case Cardano.CertificateType.PoolRetirement: {
+        if (poolIds.has(cert.poolId)) returnedDeposit += depositParams.poolDeposit;
+        break;
+      }
+    }
+  }
+
+  return {
+    deposit,
+    returnedDeposit
+  };
+};
+
+const computeConwayDeposits = (
+  certificates: Cardano.Certificate[],
+  rewardAccounts: Cardano.RewardAccount[],
+  networkId: Cardano.NetworkId
+): { deposit: Cardano.Lovelace; returnedDeposit: Cardano.Lovelace } => {
+  let deposit = 0n;
+  let returnedDeposit = 0n;
+
+  for (const cert of certificates) {
+    switch (cert.__typename) {
+      case Cardano.CertificateType.Registration:
+      case Cardano.CertificateType.StakeRegistrationDelegation:
+      case Cardano.CertificateType.VoteRegistrationDelegation:
+      case Cardano.CertificateType.StakeVoteRegistrationDelegation:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          deposit += cert.deposit;
+        break;
+      case Cardano.CertificateType.Unregistration:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          returnedDeposit += cert.deposit;
+        break;
+    }
+  }
+
+  return {
+    deposit,
+    returnedDeposit
+  };
+};
+
+/** Inspects a transaction for its deposits and returned deposits. */
+export const getDeposits = (
+  { stakeKeyDeposit, poolDeposit }: Pick<Cardano.ProtocolParameters, 'stakeKeyDeposit' | 'poolDeposit'>,
+  certificates: Cardano.Certificate[],
+  rewardAccounts: Cardano.RewardAccount[]
+): { deposit: Cardano.Lovelace; returnedDeposit: Cardano.Lovelace } => {
+  if (rewardAccounts.length === 0 || certificates.length === 0) return { deposit: 0n, returnedDeposit: 0n };
+
+  const poolIds = new Set(
+    rewardAccounts
+      .map((account) => Cardano.RewardAccount.toHash(account))
+      .map((hash) => Cardano.PoolId.fromKeyHash(hash))
+  );
+
+  const networkId = Cardano.RewardAccount.toNetworkId(rewardAccounts[0]);
+
+  const depositParams = {
+    poolDeposit: poolDeposit ? BigInt(poolDeposit) : 0n,
+    stakeKeyDeposit: BigInt(stakeKeyDeposit)
+  };
+
+  const shelleyDeposits = computeShellyDeposits(depositParams, certificates, rewardAccounts, poolIds, networkId);
+  const conwayDeposits = computeConwayDeposits(certificates, rewardAccounts, networkId);
+
+  return {
+    deposit: shelleyDeposits.deposit + conwayDeposits.deposit,
+    returnedDeposit: shelleyDeposits.returnedDeposit + conwayDeposits.returnedDeposit
+  };
+};
+
+/**
+ * Gets the collateral specified for this transaction.
+ *
+ * @param tx transaction to inspect.
+ * @param inputResolver input resolver.
+ */
+export const getCollateral = async (
```

Comment:
@renanvalentin please review this function to confirm it matches requirements for your current task

Reply:
@mkazlauskas I used @AngelCastilloB  as a base to also consider own vs foreign address. See https://github.com/input-output-hk/lace/pull/844/files#diff-3841c99e0a0dd2e29e4b14c462861db6300ba1fdd9889b82b79a901aae056096R4

Reply:
I will use @renanvalentin version of the function then. Thanks @renanvalentin

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/0f21727f0f32928b0363582ca7592d38f850cc2e